.github/workflows/pytest-gpu.yml

```diff
@@ -15,85 +11,31 @@ on:
       - master
 
 jobs:
-
-  # Deploy VM in Azure
-  deploy-vm:
-    runs-on: ubuntu-latest
+  build:
+    name: pytest-gpu
+    runs-on: [self-hosted, gpu]
 
     steps:
-    # checkout code from repo
-    - name: checkout repo
+    - name: Checkout devito
       uses: actions/checkout@v1
 
-    - name: extract branch name
-      shell: bash
+    - name: Install dependencies
       run: |
-        if [ "${{ github.head_ref }}" == "" ]; then
-          branch="master"
-        else
-          branch="${{ github.head_ref }}"
-        fi
-        echo ::set-output name=branch::$branch
-      id: extract_branch
+        pip3 install --upgrade pip
+        pip3 install -e .[extras]
 
-    - name: find deploy script
+    - name: Test with pytest
       run: |
-        ls '${{ env.OUTPUT_PATH }}/.github/azure'
-
-    - name: deploy VM
-      env:
-        RESOURCE_GROUP: ${{ secrets.CI_GPU_RESOURCE_GROUP }}
-        RESOURCE_GROUP_REGION: uksouth
-        VM_SIZE: Standard_NC6_Promo
-        SERVER_NAME: gpu_ci_vm_${{ steps.extract_branch.outputs.branch }}
-        IMAGE: gpu_ci_vm-image-1226
-      run: >
-        pwsh -command "& '${{ env.OUTPUT_PATH }}\.github\azure\deployVM.ps1'"
-        -servicePrincipal ${{ secrets.SERVICE_PRINCIPAL_APPID }}
-        -servicePrincipalSecret ${{ secrets.SERVICE_PRINCIPAL_SECRET }}
-        -servicePrincipalTenantId ${{ secrets.SERVICE_PRINCIPAL_TENANTID }}
-        -azureSubscriptionName ${{ secrets.AZURE_SUBSCRIPTION_ID }}
-        -resourceGroupName $RESOURCE_GROUP
-        -resourceGroupNameRegion $RESOURCE_GROUP_REGION
-        -size $VM_SIZE
-        -serverName $SERVER_NAME
-        -vmImage $IMAGE
-        -adminLogin ${{ secrets.CI_GPU_VM_ADMIN_LOGIN }}
-        -adminPassword ${{ secrets.CI_GPU_VM_ADMIN_PASSWORD }}
-
-    - name: set host
-      run: echo ::set-output name=action_host::$(az vm show -d -g ${{ secrets.CI_GPU_RESOURCE_GROUP }} -n gpu_ci_vm_${{ steps.extract_branch.outputs.branch }} --query publicIps -o tsv)
-      id: host
-
-    - name: execute tests on VM
-      uses: fifsky/ssh-action@master
+        # OpenMP5
+        DEVITO_ARCH=clang DEVITO_PLATFORM=nvidiaX pytest --cov --cov-config=.coveragerc --cov-report=xml tests/test_gpu_openmp.py
```

Comment:
Now that it's a normal action, can these ENV var be set outside of run? Or as a mtrix or something?

Reply:
Yes, should be able to achieve something like that.